os/board/common/crashdump.c

```diff
@@ -326,16 +326,14 @@ void print_sector(char *buf, int size) {
 	}
 }
 
-static int smartfs_dump()
+static int smartfs_dump(void)
```

Comment:
Actually flash dump is not related to smartfs.
1. First you should rename smartfs_dump to flash_dump
2. You have to check CONFIG_MTD, and something more related to MTD because we use MTD for all flashes.
3. And then you can get partition list through the procfs, CONFIG_FS_PROCFS_EXCLUDE_MTD must be enabled.
    But some code has to implmented for  procfs.
4. Now you can get information of start offset of target filesystem.
5. then you can get information about sector & block through the 'statfs', you don't have to use configs related to smartfs.